cranelift/interpreter/src/step.rs

```diff
@@ -63,7 +63,22 @@ where
     };
 
     // Retrieve the immediate value for an instruction, expecting it to exist.
-    let imm = || -> V { V::from(inst.imm_value().unwrap()) };
+    let imm = || -> V {
+        V::from(match inst {
+            InstructionData::UnaryConst {
+                constant_handle, ..
+            } => {
+                let buffer = state
+                    .get_current_function()
+                    .dfg
+                    .constants
+                    .get(constant_handle.clone())
+                    .as_slice();
+                DataValue::V128(buffer.try_into().expect("a 16-byte data buffer"))
```

Comment:
Side comment, no changes needed: yeah, wish this was easier. The decision to store the vector immediates in a side table was made to keep the size of `InstructionData` under some number of bytes. I wonder if things have changed enough since then for us to stuff 16 bytes into `InstructionData`.